src/test/java/com/ibm/cloud/sdk/core/test/service/RequestBuilderTest.java

```diff
@@ -286,6 +286,49 @@ public void testConstructHttpUrlGood() {
     assertNotEquals("https://myserver.com/testservice/api/v1/seg1/param1/seg2/param3/seg3", url);
   }
 
+  @Test
+  public void testConstructHttpUrlEmptyPath1() {
+    String[] pathSegments = { "", "discovery" };
+    HttpUrl url = RequestBuilder.constructHttpUrl("https://myserver.com/testservice/api", pathSegments);
+    assertNotNull(url);
+    assertEquals("https://myserver.com/testservice/api/discovery", url.toString());
+  }
+  
+  @Test
+  public void testConstructHttpUrlEmptyPath2() {
+    String[] pathSegments = { "" };
+    HttpUrl url = RequestBuilder.constructHttpUrl("https://myserver.com/testservice/api", pathSegments);
+    assertNotNull(url);
+    assertEquals("https://myserver.com/testservice/api", url.toString());
+  }
+
+  @Test
+  public void testConstructHttpUrlEmptyPathWParams() {
+    String[] pathSegments = { "" };
+    String[] pathParameters = { "param1", "param2" };
+    HttpUrl url = RequestBuilder.constructHttpUrl("https://myserver.com/testservice/api", pathSegments, pathParameters);
+    assertNotNull(url);
+    assertEquals("https://myserver.com/testservice/api/param1", url.toString());
```

Comment:
Could you check to see the result of this test BEFORE your fix was applied?
I'm sort of wondering whether we'd end up with a url of `https://myserver.com/testservice/api/param1/param2`.
I can imagine a path like "/{param1}/{param2}" which would result in "" within the pathSegments list, but {"param1", "param2" in the pathParameters list (I think).   And in that case I think we'd expect the url to be `https://myserver.com/testservice/api/param1/param2`, wouldn't we?
I realize this might be a corner case, but I can't help myself from always bringing those up :)

Reply:
I just tested this without my changes and the result is still the same ``https://myserver.com/testservice/api/param1``. The `/` most likely got added by the empty path segment, then the library saw that param1 didn't need a a `/` in front of it